sdjwt/src/main/kotlin/web5/security/SdJwtBlinder.kt

```diff
@@ -0,0 +1,306 @@
+  return MessageDigest.getInstance(type)
+    .digest(input)
+}
+
+/** Base class to be used for any blinding strategies. */
+public sealed class BlindOption
+
+/** FlatBlindOption implements https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-04.html#name-option-1-flat-sd-jwt */
+public object FlatBlindOption : BlindOption()
+
+/** SubClaimBlindOption implements https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-04.html#name-option-2-structured-sd-jwt */
+public class SubClaimBlindOption(
+  public val claimsToBlind: Map<String, BlindOption>
+) : BlindOption()
+
+/** RecursiveBlindOption implements https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-04.html#name-option-3-sd-jwt-with-recurs */
+public object RecursiveBlindOption : BlindOption()
+
+/** Implements https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-05.html#name-array-elements */
+public object ArrayBlindOption : BlindOption()
+
+
+/** Interface for generating salt values as described in https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-05.html#disclosures_for_object_properties */
+public interface ISaltGenerator {
+  /** Returns a base64url encoded string over cryptographically secure pseudo random data. */
+  public fun generate(claim: String): String
+}
+
+/** Simple implementation of [ISaltGenerator]. */
+public class SaltGenerator(private val numBytes: Int = 128 / 8) : ISaltGenerator {
+  override fun generate(claim: String): String {
+    val data = ByteArray(numBytes)
+    SecureRandom().nextBytes(data)
+    return Base64URL.encode(data).toString()
+  }
+}
+
+private class ClaimSetBlinder(
+  private val sdAlg: HashFunc,
+  private val disclosureFactory: DisclosureFactory,
+  private val totalDigests: (Int) -> Int,
+  private val shuffle: (List<Any>) -> Unit,
+) {
+  fun blindElementsRecursively(elems: List<Any>): Pair<List<Any>, List<Disclosure>> {
+    val blinded = mutableListOf<Any>()
+    val allDisclosures = mutableListOf<Disclosure>()
+    for (elem in elems) {
+      val claimsToBlind = mutableMapOf<String, BlindOption>()
+
+      when (elem) {
+        is Map<*, *> -> {
+          elem.keys.forEach { k ->
+            claimsToBlind[k as String] = RecursiveBlindOption
+          }
+
+          @Suppress("UNCHECKED_CAST")
+          val elemMap = elem as Map<String, Any>
+
+          val (blindedValue, ds) = toBlindedClaimsAndDisclosures(elemMap, claimsToBlind)
+          blinded.add(blindedValue)
+          allDisclosures.addAll(ds)
+        }
+
+        is List<*> -> {
+          @Suppress("UNCHECKED_CAST")
+          val elemList = elem as List<Any>
+
+          val (blindedValue, ds) = blindElementsRecursively(elemList)
+          blinded.add(blindedValue)
+          allDisclosures.addAll(ds)
+        }
+
+        else -> blinded.add(elem)
+      }
+    }
+    return Pair(blinded, allDisclosures)
+  }
+
+  fun toBlindedClaimsAndDisclosures(
```

Comment:
Can we add documentation on these functions?

Reply:
Decided against it because they're all private funcs.

Reply:
@andresuribe87 can you add comments or doc for this function? For code readability rather than external documentation